doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml

```diff
@@ -0,0 +1,119 @@
+@startuml Eclipse iceoryx Component Overview
+skinparam BackgroundColor transparent
+skinparam componentStyle uml2
+
+skinparam component {
+   backgroundColor<<namespace>> yellow
+}
+
+skinparam package {
+   backgroundColor green
+}
+
+skinparam frame {
+   backgroundColor chocolate
+}
+
+' PlantUML package: CMake package
+' PlantUML frame: CMake library
+
+' @todo #590
+' RouDi split, separate files and namespaces for static/dynamic discovery
+' Implement exported libraries (add_library in iceoryx_posh/CMakeLists.txt)
+' RouDiEnvironment shall not be in roudi namespace, but in testing
+
+header Eclipse iceoryx component overview
+
+' POSH: POSIX SHared memory
+' Publish/subscribe IPC communication infrastructure
+package iceoryx_posh {
+   frame "core library" {
+      component popo <<namespace>>
+      component capro <<namespace>>
+      component mepoo <<namespace>>
+      component version <<namespace>>
+   }
+
+   frame "runtime library" {
+      component runtime <<namespace>>
+   }
+
+   frame "gateway library" {
+      component gw <<namespace>>
+   }
+
+   frame "RouDi library" {
+      component roudi <<namespace>>
+      component build <<namespace>>
+   }
+
+   frame "config library" {
+      component config <<namespace>>
+      component helper <<namespace>>
+   }
+   component errors <<namespace>>
+}
+
+' HOOFS: Helpful Objects Optimised For Safety
+' Library containing modern C++ STL constructs
+package iceoryx_hoofs  {
+   frame "hoofs library" {
+      component concurrent <<namespace>>
```

Comment:
Is this list incomplete on purpose because it is a first draft?

Reply:
@elfenpiff This list is a draft for the structure of the nextgen iceoryx. It's still under discussion if these changes will be applied for `v2.0.0`. You can run `./tools/print_all_namespace.sh` to get a list of all namespace in iceoryx.